ext/openssl/ossl_ssl.c

```diff
@@ -1550,6 +1550,14 @@ ossl_ssl_s_alloc(VALUE klass)
     return TypedData_Wrap_Struct(klass, &ossl_ssl_type, NULL);
 }
 
+static const char*
+peeraddr_ip_str(VALUE self) {
+    VALUE peeraddr = rb_funcall(self, rb_intern("peeraddr"), 0);
+    VALUE peer_ip_entry = rb_ary_entry(peeraddr, RARRAY_LEN(peeraddr) - 1);
+
+    return RSTRING_PTR(peer_ip_entry);
```

Comment:
I'm concerned that we are returning a temporary string. Is it a problem? I don't know off the top of my head.

Reply:
I'm not sure if it's an issue. How would we avoid it?

Reply:
If the string is garbage collected before it's used, it might be an issue.